ts/Stock/Indicators/RSI/RSIIndicator.ts

```diff
@@ -106,20 +107,28 @@ class RSIIndicator extends SMAIndicator {
             change: number,
             avgGain: number,
             avgLoss: number,
-            i: number;
+            i: number,
+            values: Array<number>;
 
         // RSI requires close value
-        if (
-            (xVal.length < period) || !isArray(yVal[0]) ||
-            yVal[0].length !== 4
-        ) {
+        if ((xVal.length < period)) {
             return;
         }
+        if (!isArray(yVal[0])) {
```

Comment:
I think the `isNumber(yVal[0])` check would be simpler.

Reply:
It already was checked this way, and In my opinion, if you check if it is an array it is more readable.

Reply:
It was checked this way because, previously, it could be only an array. It made sense then, but now you changed it. Now, it can be either number or array so, if you are checking whether something is a number, then, in my humble opinion, a simple check `isNumber()` is just perfect for this. It's your call

Reply:
Ok, I reconsidered, in this case both are ok 👍

Reply:
Both are correct and I shouldn't request changes in this case. However, I disagree that it's more readable this way, because you don't indicate what type is in the first check (we only know that it's not an array).
The most readable and safe check would be:
```js
if (isNumber()) {
...
} else if (isArray() && array.length < 4) {
...
}
```

Reply:
currently this block looks like this:  
```
 if (!isArray(yVal[0])) {
            values = yVal as Array<number>;
        } else {
            // in case of the situation, where the series type has data length shorter
            // then 4 (HLC, range), this ensures that we are not trying to
            // reach the index out of bounds
            index = Math.min(params.index as number, yVal[0].length - 1);
            values = (yVal as Array<Array<number>>).map((value: Array<number>): number => value[index]);
        }

```

Reply:
so in your example it would be ```js
if (isNumber()) {
...
} else if (isArray() && array.length < 4) {
...
} else{ //??????
}
```